cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/RackAwareGoal.java

```diff
@@ -206,14 +215,18 @@ protected void initGoalState(ClusterModel clusterModel, OptimizationOptions opti
   @Override
   protected void updateGoalState(ClusterModel clusterModel, Set<String> excludedTopics)
       throws OptimizationFailureException {
-    // One pass is sufficient to satisfy or alert impossibility of this goal.
-    // Sanity check to confirm that the final distribution is rack aware.
-    ensureRackAware(clusterModel, excludedTopics);
-    // Sanity check: No self-healing eligible replica should remain at a dead broker/disk.
-    GoalUtils.ensureNoOfflineReplicas(clusterModel, name());
-    // Sanity check: No replica should be moved to a broker, which used to host any replica of the same partition on its broken disk.
-    GoalUtils.ensureReplicasMoveOffBrokersWithBadDisks(clusterModel, name());
-    finish();
+    try {
+      // One pass is sufficient to satisfy or alert impossibility of this goal.
+      // Sanity check to confirm that the final distribution is rack aware.
+      ensureRackAware(clusterModel, excludedTopics);
+      // Sanity check: No self-healing eligible replica should remain at a dead broker/disk.
+      GoalUtils.ensureNoOfflineReplicas(clusterModel, name());
+      // Sanity check: No replica should be moved to a broker, which used to host any replica of the same partition on its broken disk.
+      GoalUtils.ensureReplicasMoveOffBrokersWithBadDisks(clusterModel, name());
+      finish();
+    } finally {
+      clusterModel.clearSortedReplicas();
```

Comment:
Do we have a unit test to verify that the sortedReplicas are always cleared after each goal? (if not can we add one)?

Reply:
I changed the location calling this method, so that it is easy to reason that each goal will call it unconditionally.

Reply:
Thanks for the changes! Can we also add an `assert` / `validation` regarding whether goal states are cleared after execution?